### ISSUE TITLE
Use Docker to encapsulate environments

### ISSUE THREAD

**Issue by drauschenbach** [andreaferretti/kmeans#58]
Andrea, this is a neat project. I wonder if adding a Dockerfile to each directory would make this project fun for you again, and relieve you of having to manually manage a polyglot host environment. If all tests are run in a container, it would be a fairly level playing field, and the results would still work as a relative comparison.

**Comment by andreaferretti** [comment]
Yeah, that would probably work, although I happen to prefer [Vagga](https://github.com/tailhook/vagga).

What would **not** be fun is writing the dockerfiles for all languages! :-)

**Comment by drauschenbach** [comment]
Vagga is Linux-only and this project has a much broader audience than that.

I think writing the dockerfiles sounds fun! I'm sure you could crowdsource most of it by creating the issues & letting people volunteer.

**Comment by andreaferretti** [comment]
Docker is linux only as well, it just happens to have a convenient way to run it in a virtual machine (but [vagga has one too](https://github.com/tailhook/vagga-box), although less developed).

But I digress - anyway, I welcome the contribution of dockerfiles, I just do not have time to write them all myself! If you want to contribute a few, please just open a PR :-)

**Comment by drauschenbach** [comment]
Awesome! OK great.
